### PR TITLE
Wish list functionality

### PR DESCRIPTION
**Pull request by Ankur9669**
Files to check:
1. WishList.js
2. wishlist-context.js
3. wishlist-reducer.js
4. add-to-wishlist.js
5. remove-from-wishlist.js

### CODE REVIEW

File: fitbit-react/src/Reducers/wishlist-reducer.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const wishListReducer = (wishlist, action) => {
+  switch (action.type) {
+    case "SET_WISHLIST":
+      return [...action.payload.value];
+    default:
+      return wishlist;
+  }
+};
+export { wishListReducer };
```
Comment (bhtibrewal) [inline fitbit-react/src/Reducers/wishlist-reducer.js:9]: Suggestion you could have also managed this with only useState as it doesn't involve big complex logic

File: fitbit-react/src/index.js [hunk 2]
```diff
@@ -18,9 +19,11 @@ ReactDOM.render(
...
-              <App />
-            </CartProvider>
+            <WishListProvider>
+              <CartProvider>
+                <App />
+              </CartProvider>
+            </WishListProvider>
           </FilterProvider>
         </ProductProvider>
       </UserProvider>
```
Comment (bhtibrewal) [inline fitbit-react/src/index.js:29]: Because there are so many contexts you can combine them into one

**Review by bhtibrewal — APPROVED** [review]
Some small suggestions, else everything is okay good to merge

File: fitbit-react/src/Components/VerticalCard/VerticalCard.js [hunk 3]
```diff
@@ -31,9 +39,12 @@ function VerticalCard({ product }) {
...
 
   // Function to check if product exists in cart
-  const findIfProductExistInCart = (productId) => {
-    return cart.some((cartItem) => cartItem._id === productId);
-  };
+  // const findIfProductExistInCart = (productId) => {
+  //   return cart.some((cartItem) => cartItem._id === productId);
+  // };
+
+  let ifProductExistInCart = findIfProductExistsInArray(cart, _id);
```
Comment (sri-rishi) [inline fitbit-react/src/Components/VerticalCard/VerticalCard.js]: Suggestion: if value is not changing, you can use `const` variable naming

File: fitbit-react/src/Components/VerticalCard/VerticalCard.js [hunk 4]
```diff
@@ -49,7 +60,29 @@ function VerticalCard({ product }) {
...
+                        wishList = await addToWishList(product);
+                      } else {
+                        wishList = await removeFromWishList(_id);
+                      }
+                      dispatchWishList({
+                        type: "SET_WISHLIST",
+                        payload: { value: wishList.wishlist },
+                      });
+                    }
+                  : () => navigate("/login")
```
Comment (sri-rishi) [inline fitbit-react/src/Components/VerticalCard/VerticalCard.js]: Suggestion: you can define function somewhere and use it here

File: fitbit-react/src/Pages/Wishlist/WishList.js [hunk 5]
```diff
@@ -12,7 +13,7 @@ function WishList() {
         <h3 class="h2 centered-text">My WishList</h3>
```
Comment (sri-rishi) [inline fitbit-react/src/Pages/Wishlist/WishList.js]: Suggestion: I think you forgot to change `class` to  `className`

**Review by sri-rishi — APPROVED** [review]
Code looks fine, just some changes which I think can be done are mentioned below

File: fitbit-react/src/Components/VerticalCard/VerticalCard.js (hunk 3 above)
Comment (Ankur9669) [reply fitbit-react/src/Components/VerticalCard/VerticalCard.js]: Done

File: fitbit-react/src/Components/VerticalCard/VerticalCard.js (hunk 4 above)
Comment (Ankur9669) [reply fitbit-react/src/Components/VerticalCard/VerticalCard.js]: Done

File: fitbit-react/src/index.js (hunk 2 above)
Comment (Ankur9669) [reply fitbit-react/src/index.js:29]: Will try that

File: fitbit-react/src/Reducers/wishlist-reducer.js (hunk 1 above)
Comment (Ankur9669) [reply fitbit-react/src/Reducers/wishlist-reducer.js:9]: Ya i can but for consistency i preffered this

File: fitbit-react/src/Pages/Wishlist/WishList.js (hunk 5 above)
Comment (Ankur9669) [reply fitbit-react/src/Pages/Wishlist/WishList.js]: Done

**Pull request merged**